src/status_im/contexts/wallet/sheets/slippage_settings/view.cljs

```diff
@@ -0,0 +1,115 @@
+(ns status-im.contexts.wallet.sheets.slippage-settings.view
+  (:require [clojure.string :as string]
+            [quo.core :as quo]
+            [react-native.core :as rn]
+            [status-im.constants :as constants]
+            [status-im.contexts.wallet.sheets.slippage-settings.style :as style]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn- validate-slippage
+  [slippage]
+  (let [slippage-value (js/parseFloat slippage)]
+    (cond
+      (<= slippage-value 0)                     {:message (i18n/label
+                                                           :t/slippage-should-be-more-than-0)
+                                                 :type    :error}
+      (> (count (second (string/split slippage ".")))
+         constants/max-slippage-decimal-places) {:message (i18n/label :t/max-2-decimals)
+                                                 :type    :error}
+      (> slippage-value
+         constants/max-recommended-slippage)    {:message (i18n/label
+                                                           :t/slippage-may-be-higher-than-necessary)
+                                                 :type    :warning}
+      :else                                     nil)))
+
+(defn- on-cancel
+  []
+  (rf/dispatch [:hide-bottom-sheet]))
+
+(defn view
+  []
+  (let [current-slippage                (rf/sub [:wallet/swap-max-slippage])
+        account-color                   (rf/sub [:wallet/current-viewing-account-color])
+        [max-slippage set-max-slippage] (rn/use-state (str current-slippage))
+        [error set-error]               (rn/use-state nil)
+        [custom? set-custom?]           (rn/use-state (not-any? #{current-slippage}
+                                                                constants/slippages))
+        handle-slippage-change          (rn/use-callback
+                                         (fn [value]
+                                           (let [new-slippage (if (= value :i/backspace)
```

Comment:
You could extract this logic into a separate function var, for example `join-or-truncate-last-char`, or of course any other name you prefer.

```clojure
(defn- join-or-truncate-last-char
  [maybe-key s]
  (if (= :i/backspace maybe-key)
    (subs s 0 (dec (count s)))
    (str s maybe-key)))
```